### PR TITLE
added automatic publication of npm package

### PR DESCRIPTION
**Pull request by baldarn**

### CODE REVIEW

**Comment by baldarn** [comment]
TODO: before merging the secret `${{secrets.NPM_TOKEN}}` must be set ;)

File: package.json [hunk 1]
```diff
@@ -10,7 +10,9 @@
     "check-ts": "tsc --noEmit && tsc --project test/tsconfig.json --noEmit",
     "check-lint": "tsc --noEmit && eslint 'src/**/*.ts' 'test/**/*.ts'",
     "check-format": "prettier -c 'src/**/*.ts' 'test/**/*.ts'",
-    "check-spell": "cspell 'src/**/*ts' 'test/**/*ts'"
+    "check-spell": "cspell 'src/**/*ts' 'test/**/*ts'",
+    "types:prepare": "rm -rf dist && tsc --declaration true --sourceMap false && cp package.json dist/package.json && cp README.md dist/README.md && cd dist && npm version $(git describe --tags --always)",
```
Comment (gpad) [inline package.json:14]: Insetad of `tsc --declaration true --sourceMap false` why don't use the build?

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
...
+    "types:publish": "cd dist && npm publish --access public"
```
Comment (gpad) [inline package.json:15]: I have always published the entire package not only dist. Are you sure it's a good idea publish only `dist`

**Comment by gpad** [comment]
What do you think about this PR https://github.com/coders51/rabbitmq-stream-js-client/pull/41

File: .github/workflows/release-package.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+name: Release Package
+
+on:
+  release:
+    types: [created]
+  push:
+    branches:
+      - main
```
Comment (gpad) [inline .github/workflows/release-package.yml:8]: @baldarn I think in this way every time we merge on main we publish a new version, I don't think it's correct.

**Review by gpad — APPROVED** [review]

**Pull request merged**